sycl/test-e2e/Matrix/joint_matrix_rowmajorA_rowmajorB_impl.hpp

```diff
@@ -0,0 +1,146 @@
+         }); // parallel for
+   }).wait();
+}
+
+template <size_t TN, size_t TK, class kernel_name, typename TA, typename TB,
+          typename TC>
+int gemm_row_major() {
+  static constexpr size_t TM = 8;
+
+  static constexpr size_t MATRIX_M = TM * 2;
+  static constexpr size_t MATRIX_N = TN * 2;
+  static constexpr size_t MATRIX_K = TK * 2;
+  TA A[MATRIX_M][MATRIX_K];
+  TB B[MATRIX_K][MATRIX_N];
+  TC C[MATRIX_M][MATRIX_N];
+  TC D[MATRIX_M][MATRIX_N];
+
+  matrix_fill(MATRIX_M, MATRIX_K, (TA *)A,
+              [](int i, int j) { return 1 * (i + j); });
+  matrix_fill(MATRIX_K, MATRIX_N, (TB *)B,
+              [](int i, int j) { return 2 * i + 3 * j; });
+  matrix_fill(MATRIX_M, MATRIX_N, (TC *)C, (TC)1);
+  matrix_fill(MATRIX_M, MATRIX_N, (TC *)D, (TC)1);
+
+  big_matrix<TC, MATRIX_M, MATRIX_N> MC((TC *)&C);
+  big_matrix<TC, MATRIX_M, MATRIX_N> MD((TC *)&D);
+  big_matrix<TA, MATRIX_M, MATRIX_K> MA((TA *)&A);
+  big_matrix<TB, MATRIX_K, MATRIX_N> MB((TB *)&B);
+  matrix_multiply<TM, TN, TK, kernel_name>(MC, MA, MB);
+  matrix_multiply_ref((TA *)A, (TB *)B, (TC *)D, MATRIX_M, MATRIX_N, MATRIX_K);
+
+  bool res = matrix_compare(MATRIX_M, MATRIX_N, (TC *)C, (TC *)D);
+  std::cout << (res ? "passed" : "failed") << std::endl;
+  return !res;
+}
+
+int main() {
+  queue q;
+  std::vector<combination> combinations =
+      q.get_device()
+          .get_info<sycl::ext::oneapi::experimental::info::device::
+                        matrix_combinations>();
+  for (unsigned int i = 0; i < combinations.size(); i++) {
+    if (combinations[i].atype == matrix_type::bf16) {
+      if (combinations[i].nsize == 0 || combinations[i].nsize == 16) {
+        gemm_row_major<16, 16, class gemm_bfloat16_16, bfloat16, bfloat16,
+                       float>();
+        break;
+      }
+      if (combinations[i].nsize == 8) {
+        gemm_row_major<8, 16, class gemm_bfloat16_8, bfloat16, bfloat16,
+                       float>();
+        break;
+      }
+    }
+    if (combinations[i].atype == matrix_type::sint8) {
+      if (combinations[i].nsize == 0 || combinations[i].nsize == 16) {
+        gemm_row_major<16, 32, class gemm_int8_16, int8_t, int8_t, int32_t>();
+        gemm_row_major<16, 32, class gemm_us_int8_16, uint8_t, int8_t,
+                       int32_t>();
+        gemm_row_major<16, 32, class gemm_su_int8_16, int8_t, uint8_t,
+                       int32_t>();
+        gemm_row_major<16, 32, class gemm_uu_int8_16, uint8_t, uint8_t,
+                       int32_t>();
+        break;
+      }
+      if (combinations[i].nsize == 8) {
+        gemm_row_major<8, 32, class gemm_int8_8, int8_t, int8_t, int32_t>();
+        gemm_row_major<8, 32, class gemm_us_int8_8, uint8_t, int8_t, int32_t>();
+        gemm_row_major<8, 32, class gemm_su_int8_8, int8_t, uint8_t, int32_t>();
+        gemm_row_major<8, 32, class gemm_uu_int8_8, uint8_t, uint8_t,
+                       int32_t>();
+        break;
+      }
+    }
+  }
```

Comment:
Breaks should be removed. Otherwise, if test runs for one type (e.g. bf16), it will not run for the other type (e.g. sint8)

```suggestion
  for (unsigned int i = 0; i < combinations.size(); i++) {
    if (combinations[i].atype == matrix_type::bf16) {
      if (combinations[i].nsize == 0 || combinations[i].nsize == 16) {
        gemm_row_major<16, 16, class gemm_bfloat16_16, bfloat16, bfloat16,
                       float>();
      }
      if (combinations[i].nsize == 8) {
        gemm_row_major<8, 16, class gemm_bfloat16_8, bfloat16, bfloat16,
                       float>();
      }
    }
    if (combinations[i].atype == matrix_type::sint8) {
      if (combinations[i].nsize == 0 || combinations[i].nsize == 16) {
        gemm_row_major<16, 32, class gemm_int8_16, int8_t, int8_t, int32_t>();
        gemm_row_major<16, 32, class gemm_us_int8_16, uint8_t, int8_t,
                       int32_t>();
        gemm_row_major<16, 32, class gemm_su_int8_16, int8_t, uint8_t,
                       int32_t>();
        gemm_row_major<16, 32, class gemm_uu_int8_16, uint8_t, uint8_t,
                       int32_t>();
      }
      if (combinations[i].nsize == 8) {
        gemm_row_major<8, 32, class gemm_int8_8, int8_t, int8_t, int32_t>();
        gemm_row_major<8, 32, class gemm_us_int8_8, uint8_t, int8_t, int32_t>();
        gemm_row_major<8, 32, class gemm_su_int8_8, int8_t, uint8_t, int32_t>();
        gemm_row_major<8, 32, class gemm_uu_int8_8, uint8_t, uint8_t,
                       int32_t>();
      }
    }
  }
```

Reply:
Yes, exactly! :)